chapters/mitigations-and-defensive-strategies/defense-mechanisms/activities/05-rwslotmachine5/sol/Dockerfile

```diff
@@ -0,0 +1,29 @@
+# --- Build ---
+FROM gcc:latest AS builder
+
+# Set working directory
+WORKDIR /app
+
+# Copy the source code
+COPY ./rwslotmachine5.c .
+
+# Build the C program
+RUN gcc -o rwslotmachine5 rwslotmachine5.c
+
+# --- Runtime ---
+FROM ubuntu:22.04
+
+# Install minimal libraries needed for runtime
+RUN apt-get update && apt-get install -y libstdc++6 && rm -rf /var/lib/apt/lists/*
+
+# Set working directory
+WORKDIR /app
+
+# Copy compiled binary from builder stage
+COPY --from=builder /app/rwslotmachine5 .
+
+# Expose port
+EXPOSE 31348
+
+# Run the executable
+CMD ["/app/rwslotmachine5"]
```

Comment:
There are too many comments. Some of the commands are self-explanatory, remove those comments.